### DISCUSSION TITLE
How to use asr model for streaming

### DISCUSSION THREAD

**Discussion by ravi02512** [NVIDIA-NeMo/NeMo#2331] [Q&A] [answered]
Hello folks

I used nvidia quartznet for asr and it is taking wav files to transcribe. How to do real time streaming asr model

Thanks

**Comment by snaaz21** [comment]
How to do real time streaming using Citrinet asr model? Will it be the same way as of Quartznet?

**Comment by ravi02512** [comment]
> How to do real time streaming using Citrinet asr model? Will it be the same way as of Quartznet?

do you know how to implement  quartznet for real time streaming??

**Comment by snaaz21** [comment]
> > How to do real time streaming using Citrinet asr model? Will it be the same way as of Quartznet?
> 
> do you know how to implement quartznet for real time streaming??

No, I didn't try yet by myself.

**Comment by khursani8** [comment, marked as answer]
For streaming you need FrameAsr like from this notebook https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/02_Online_ASR_Microphone_Demo.ipynb or use Nvidia Jarvis

**Comment by snaaz21** [comment]
> For streaming you need FrameAsr like from this notebook https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/02_Online_ASR_Microphone_Demo.ipynb or use Nvidia Jarvis

Yes, I tried this earlier but it didn't work for CitriNet model.

**Comment by khursani8** [comment]
What you mean by didn't work? Did you use correct stats? https://github.com/NVIDIA/NeMo/issues/1203
And from what I know CitriNet use different stride,you might need to take that into account, I don't have experience with CitriNet yet

**Comment by snaaz21** [comment]
> What you mean by didn't work? Did you use correct stats? #1203
> And from what I know CitriNet use different stride,you might need to take that into account, I don't have experience with CitriNet yet

Correct, It is working for Quartznet model, but no output was coming for Citrinet model. I wanted to know what could be the reason of it. @khursani8 thanx for your response, I'll be checking for different stride.

**Comment by okuchaiev** [comment]
* For example on how to use QuartzNet in streaming mode (with microphone) please refer to: https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/02_Online_ASR_Microphone_Demo.ipynb
* If you need to transcribe long audio file refer to this discussion (CitriNet) https://github.com/NVIDIA/NeMo/discussions/2172 

We are working on adding streaming ASR tutorial. @jbalam-nv fyi.

**Comment by ridasaleem0** [comment]
@snaaz21 were you able to run Citrinet model for asr (with microphone)? Can you guide a bit, would be huge help.

**Comment by arunvenkatesan-nv** [comment]
Streaming tutorial that should work with Both Conformer-CTC and Citrinet are now in NeMo: https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/Streaming_ASR.ipynb

This is also available as a script for offline long form audio decoding: https://github.com/NVIDIA/NeMo/blob/main/examples/asr/speech_to_text_buffered_infer.py
